### PR TITLE
Just-in-Time Hint Prompt

### PR DESCRIPTION
**Pull request by Hamms**
[spec](https://docs.google.com/document/d/10-pBXvMtP2j-iRfTTCjSCYoHkfCeBVwgHr2z0VF9x9s/edit#heading=h.7zszt1vt37v3)

Depends on https://github.com/code-dot-org/code-dot-org/pull/17363 to be merged and the associated script to be run in order for thresholds to be set accurately

### CODE REVIEW

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -1433,14 +1435,21 @@ StudioApp.prototype.displayFeedback = function (options) {
...
+    );
     this.displayMissingBlockHints(missingBlockHints);
 
     // communicate the feedback message to the top instructions via
     // redux
     const message = this.feedback_.getFeedbackMessage(options);
     getStore().dispatch(setFeedback({ message }));
   }
+
+  if (experiments.isEnabled('onetimeHintPrompt')) {
```
Comment (Hamms) [inline apps/src/StudioApp.js]: @ryansloan do we want this behind an experiments flag initially?

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -460,6 +460,8 @@
   "online": "Online",
   "openWorkspace": "How It Works",
   "orientationLock": "Turn off orientation lock in device settings.",
+  "onetimeHintPromptTitle": "Feeling Stuck?",
+  "onetimeHintPromptMessage": "Click the lightbulb for a hint",
```
Comment (Hamms) [inline apps/i18n/common/en_us.json]: @ryansloan there's still an open question in the spec if this is the copy we actually want to go with

Comment (ryansloan) [reply apps/i18n/common/en_us.json]: Go with this copy for now. We are also hoping to help discoverability of the lightbulb with this so I think explicitly spelling it out is good to start.

File: apps/src/StudioApp.js (hunk 1 above)
Comment (ryansloan) [reply apps/src/StudioApp.js]: Nope! :shipit:

**Comment by Hamms** [comment]
:+1: will do in a separate PR

File: apps/src/utils.js [hunk 3]
```diff
@@ -394,6 +405,19 @@ export function trySetLocalStorage(item, value) {
 
 }
 
+export function tryGetSessionStorage(key, defaultValue) {
+  if (defaultValue === undefined) {
+    throw "tryGetSessionStorage requires defaultValue";
+  }
+  let returnValue = defaultValue;
+  try {
+    returnValue = sessionStorage.getItem(key);
```
Comment (balderdash) [inline apps/src/utils.js:414]: I would expect this function to return `defaultValue` if there is no item for `key`, but this just returns `null`, right? I don't think it throws an exception for missing keys.

Comment (Hamms) [reply apps/src/utils.js:414]: correct, and I agree that it's a bit weird but I wanted to follow the pattern of [`tryGetLocalStorage`](https://github.com/code-dot-org/code-dot-org/pull/17220/files#diff-85c58ca2effd283f2dfd4479a8777179R378) which is operating the same way

**Pull request merged**